src/test/java/test/tesng1046/TestClassSample.java

```diff
@@ -0,0 +1,52 @@
+package test.tesng1046;
+
+import org.testng.IAlterTestName;
+import org.testng.IHookCallBack;
+import org.testng.IHookable;
+import org.testng.ITestResult;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+public class TestClassSample implements IHookable {
+
+    @DataProvider (name = "dp", parallel = true)
+    public Object[][] getTestData() {
+        return new Object[][] {
+            {1}, {2}, {3}, {4}, {5}
+        };
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample1(int num) {
+        assertTrue(num != 0);
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample2(int num) {
+        assertTrue(num != 0);
+    }
+
+    @Test
+    public void ordinaryTestMethod() {
+
+    }
+
+    @Override
+    public void run(IHookCallBack callBack, ITestResult testResult) {
+        Object[] parameters = callBack.getParameters();
+        String testName = name(testResult.getMethod().getMethodName(), "999");
+        if (parameters != null && parameters.length != 0) {
+            testName = name(testResult.getMethod().getMethodName(), parameters[0]);
+        }
+        if (testResult instanceof IAlterTestName) {
+            ((IAlterTestName) testResult).setTestName(testName);
+        }
+        callBack.runTestMethod(testResult);
+    }
+
+    private String name(String prefix, Object count) {
```

Comment:
Lol! I forgot. Somehow this idea is firmly stuck to my mind that `static` anything is generally an evil thing. So without even realising I stay away from it as much as possible. I remember @cbeust advice on this (in terms of calling out explicitly to the user that `hey I am not a method that depends on the state..` for better clarity). So will fix this.